src/Controllers/PaymentsForwardingController.php

```diff
@@ -19,18 +42,36 @@ public function getCallback(Request $request)
     }
 
     /**
-     * Process callback requests
-     *
+     * @param TransactionModel $transaction
      * @param Request $request
-     *
-     * @return \Illuminate\Http\Response
-     * TODO it should process requests
+     * @return Response
      */
-    public function postCallback(Request $request)
+    public function postCallback(TransactionModel $transaction, Request $request): Response
```

Comment:
We should not specify type of the return here as an implementation class. It should not be here in any way, but if you would like to do that - use the interfaces for that purpose.